app/controllers/newsletter_integration/newsletter_subscription_controller.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module NewsletterIntegration
+  class NewsletterSubscriptionController < ::ApplicationController
+    requires_plugin NewsletterIntegration::PLUGIN_NAME
+    requires_login
+
+    before_action :ensure_plugin_configured
+
+    def subscribe
+      subscription =
+        NewsletterUserSubscription.find_or_initialize_by(
+          user_id: current_user.id,
+          newsletter_integration_id: GLOBAL_NEWSLETTER_ID,
+        )
+      subscription.ensure_active!
+      render json: { message: "OK" }, status: 200
```

Comment:
This is unexpected, we have `success_json` we usually use for this, which will output: `{ success: "OK" }`

Reply:
Done!